### ISSUE TITLE
Support braces for no-fallthrough

### ISSUE THREAD

**Issue by cowwoc** [eslint/eslint#7889]
**What version of ESLint are you using?**
3.0.1
**What rule do you want to change?**
no-fallthrough
**What code should be flagged as correct with this change?**

```
/*eslint no-fallback: ["error", { "commentPattern": "[\\w\\s]*fallthrough" }]*/
switch(foo) {
    case 0: {
      if (typeof(input) == 'string')
        return something();
      // Otherwise, fallthrough
    }
    default: c();
}
```

**What happens when the rule is applied to this code now?**
Errors:
Expected a 'break' statement before 'default'.

**Comment by platinumazure** [comment]
Seems reasonable to me. 

Just to make sure I understand your proposal, using ESTree terms:

1. For a given SwitchCase,
1. If its consequent contains one statement and that is a BlockStatement,
1. Then the rule logic should effectively be run on the BlockStatement instead of the SwitchCase, and a `// falls through` label or similar at the end of the BlockStatement should be acceptable and not cause a warning. 

Does that sound correct? (Note: I'm basing this on the AST seen [here](http://astexplorer.net/#/kqpPTcTzGA).)

**Comment by cowwoc** [comment]
@platinumazure Yes, that sounds about right :) Thank you.

**Comment by cowwoc** [comment]
@not-an-aardvark I didn't get the feeling that there was a lack of consensus over this issue. Please see the last 2 comments.

**Comment by ilyavolodin** [comment]
@cowwoc what @not-an-aardvark means is consensus from the team. Every proposed change has to be supported by 3 team members and championed by one. Unfortunately, in this case, none of the team members elected to support this change other then @platinumazure 
As @not-an-aardvark message mentions, we found that if the issue doesn't gain support shortly after being created, it's most likely will be forgotten and will never gain support required for implementation:-( That's why we periodically close older issues.

**Comment by cowwoc** [comment]
Fair enough. Thanks for the clarification. Hopefully this will be revisited in the future.

**Comment by j-f1** [comment]
This has been brought up again in #9080.
